services/api/src/resources/fact/resolvers.ts

```diff
@@ -82,19 +82,24 @@ export const addFacts = async (
   { sqlClient, hasPermission }
 ) => {
 
-  // We first check that the user has access to all of the environments, so this is an atomic operation.
-  await facts.map(async (fact) => {
-    const { environment } = fact;
-    const env = await environmentHelpers(sqlClient).getEnvironmentById(environment);
+  const environments = facts.reduce((environmentList, fact) => {
+    let { environment } = fact;
+    if(!environmentList.includes(environment)) {
+      environmentList.push(environment)
+    }
+    return environmentList
+  }, []);
 
+  for(let i = 0; i < environments.length; i++) {
+    const env = await environmentHelpers(sqlClient).getEnvironmentById(environments[i]);
     await hasPermission('fact', 'add', {
       project: env.project,
     });
-  });
-
-  return await facts.map(async (fact) => {
-    const { environment, name, value, source, description } = fact;
+  }
 
+  const returnFacts = []
+  for(let i = 0; i < facts.length; i++) {
```

Comment:
These don't conform to our code style standards. We don't use ASI and use spaces after control words.

I already know that on `master` we're pretty bad on this issue, so this is just a "for next time" comment. With VS Code and the "Prettier - Code formatter" plugin you can run "Format Document" on any file to correct style issues.

Reply:
Hey @rocketeerbkw - no worries, let me fix this :)
I won't reformat the whole file since that's offsides when it comes to making actual changes - but I'll fix these up.

As for the ASI ... ahem, that was less a conscious choice than it is a consequence of writing golang recently. Sorry :)